pennylane/transforms/decompositions/pauli_based_computation.py

```diff
@@ -0,0 +1,593 @@
+# Copyright 2025 Xanadu Quantum Technologies Inc.
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#     http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Aliases for pauli-based computation passes from Catalyst's passes module."""
+
+from functools import partial
+
+from pennylane.transforms.core import transform
+
+
+@partial(transform, pass_name="to-ppr")
+def to_ppr(tape):
+    r"""A quantum compilation pass that converts Clifford+T gates into Pauli Product Rotation (PPR)
+    gates.
+
+    .. note::
+
+        This transform requires decorating the QNode with :func:`@qml.qjit <pennylane.qjit>` and for
+        program capture to be enabled via :func:`qml.capture.enable() <pennylane.capture.enable>`.
+
+    Clifford gates are defined as :math:`\exp(-{iP\tfrac{\pi}{4}})`, where :math:`P` is a Pauli word.
+    Non-Clifford gates are defined as :math:`\exp(-{iP\tfrac{\pi}{8}})`.
+
+    For more information on the PPM compilation pass, check out the
+    `compilation hub <https://pennylane.ai/compilation/pauli-product-measurement>`__.
+
+    .. note::
+
+        The circuits that generated from this pass are currently not executable on any backend.
+        This pass is only for analysis with the ``null.qubit`` device and potential future execution
+        when a suitable backend is available.
+
+    The full list of supported gates and operations are
+    ``qml.H``,
+    ``qml.S``,
+    ``qml.T``,
+    ``qml.X``,
+    ``qml.Y``,
+    ``qml.Z``,
+    ``qml.adjoint(qml.S)``,
+    ``qml.adjoint(qml.T)``,
+    ``qml.CNOT``, and
+    ``catalyst.measure``.
+
+    Args:
+        fn (QNode): QNode to apply the pass to
+
+    Returns:
+        :class:`QNode <pennylane.QNode>`
+
+    **Example**
+
+    The ``to_ppr`` compilation pass can be applied as a dectorator on a QNode:
+
+    .. code-block:: python
+
+        import pennylane as qml
+
+        qml.capture.enable()
+
+        @qml.qjit(target="mlir")
+        @qml.transforms.to_ppr
+        @qml.qnode(qml.device("null.qubit", wires=2))
+        def circuit():
+            qml.H(0)
+            qml.CNOT([0, 1])
+            qml.T(0)
+            return qml.expval(qml.Z(0))
+
+    For clear and inspectable results, use ``target="mlir"`` in the ``qjit`` decorator, ensure that
+    PennyLane's program capture is enabled, :func:`pennylane.capture.enable`, and call ``to_ppr``
+    from the PennyLane frontend (``qml.transforms.to_ppr``) instead of with
```

Comment:
Why is the `qml.trnasforms.to_ppr` more inspectable than the catalyst version? That's really weird.